services/blockchain-indexer/shared/dataService/business/transactionsEstimateFees.js

```diff
@@ -244,6 +246,36 @@ const calcAdditionalFees = async (transaction) => {
 	return additionalFees;
 };
 
+const getTransactionParamsSchema = (transaction, metadata) => {
+	const moduleMetadata = metadata.modules.find(m => m.name === transaction.module);
+	const { params: schema } = moduleMetadata.commands.find(c => c.name === transaction.command);
+	return schema;
+};
+
+const validateTransactionParams = async transaction => {
+	// Mock optional values if not present before schema validation.
+	if (transaction.module === MODULE.TOKEN
+		&& transaction.command === COMMAND.TRANSFER_CROSS_CHAIN) {
+		if (!('messageFee' in transaction.params)) {
+			transaction.params.messageFee = '10000000';
+		}
+
+		if (!('messageFeeTokenID' in transaction.params)) {
+			transaction.params.messageFeeTokenID = '0000000000000000';
+		}
+	}
+
+	const metadata = await getSystemMetadata();
+	const txParamsSchema = getTransactionParamsSchema(transaction, metadata);
```

Comment:
Instead of calling system metadata, we can call `getSchemas` method from `services/blockchain-indexer/shared/dataService/business/schemas.js`. In that case we can just do something like:
```
const allSchemas = await getSchemas();
const {txParamsSchema: schema} = allSchemas.data.commands.find(e => e. moduleCommand === `${transaction.module}:${transaction.command}`);
```